### PR TITLE
Feature 268 simply login

### PR DESCRIPTION
**Pull request by marckraw**
Not to merge, yet. Branch will be broader, because we have to make register confirm, and after confirmation views for react registration, and logic for that.

### CODE REVIEW

**Comment by marckraw** [comment]
Removing WIP label, cause it's kinda ready to be merged :) There are protected routes and others.

File: src/auth/epics/index.js [hunk 1]
```diff
@@ -19,6 +20,19 @@ export function authEpicFactory() {
...
+            ofType(actions.REGISTER_USER_CONFIRM),
+            switchMap(action =>
+                authApi
+                    .registerUserConfirm(action.payload)
+                    .catch(actions.registerUserConfirmRejected),
+            ),
+            map(res => {
+                return actions.registerUserConfirmFulfilled();
+            }),
+        );
```
Comment (tomaszferens) [inline src/auth/epics/index.js]: This won't work as expected. If registerUserConfirm will throw an error, the action `registerUserConfirmRejected` will be `mapped` to `registerUserConfirmFulfilled`. Basically: `registerUserConfirmRejected` will never run.

This should work:
```
    const registerUserConfirm = action$ =>
        action$.pipe(
            ofType(actions.REGISTER_USER_CONFIRM),
            switchMap(action =>
                authApi
                    .registerUserConfirm(action.payload)
                    .then(actions.registerUserConfirmFulfilled)
                    .catch(actions.registerUserConfirmRejected),
            ),
        );
```

File: src/auth/epics/index.js [hunk 2]
```diff
@@ -42,10 +56,36 @@ export function authEpicFactory() {
             switchMap(action =>
                 authApi
                     .loginUser(action.payload)
-                    .then(actions.loginUserFulfilled)
+                    .then(() => {
+                        localStorage.setItem('authenticated', 'true');
```
Comment (tomaszferens) [inline src/auth/epics/index.js]: I think that isAuthenticated should be in redux. Probably in `auth` state

Comment (marckraw) [reply src/auth/epics/index.js]: yup I think so, but for that I have to handle this `redux-persist` think

File: src/auth/reducers/index.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
-};
+const initialState = { authenticated: false };
 
 export function auth(state = initialState, action) {
     switch (action.type) {
         case actions.LOGIN_USER_FULFILLED:
             return {
                 ...state,
-                isLoggedIn: true,
+                authenticated: true,
```
Comment (tomaszferens) [inline src/auth/reducers/index.js:10]: oh, it already is here. Then I don't see a point in storing this information in localStorage

Comment (marckraw) [reply src/auth/reducers/index.js:10]: Yea it is there, but when refreshing it will be lost, I will handle `redux-persist` for that.

**Review by tomaszferens — APPROVED** [review]

**Comment by marckraw** [comment]
Authenticated check method change to checking with redux, also added `redux-persist` so it will be fully working even after refresh.

File: src/auth/epics/index.js (hunk 1 above)
Comment (marckraw) [reply src/auth/epics/index.js]: yup, it's still not implemented, just copied. Will change then ;)

**Pull request merged**